File: CodeBusters.py
import sys
import math
import numpy
# Send your busters out into the fog to trap ghosts and bring them home!
def dist (a,b):
    return int(math.sqrt((a[1]-b[1])**2+(a[2]-b[2])**2))
busters_per_player = int(input())  # the amount of busters you control
ghost_count = int(input())  # the amount of ghosts on the map
my_team_id = int(input())  # if this is 0, your base is on the top left of the map, if it is one, on the bottom right
my_buster=[]
ghost=[]
if my_team_id==0:
    home_x=home_y=0
    p_x=16001
    p_y=9001
else:
    home_x=16001
    home_y=9001
    p_x=p_y=0
entity_id=0# 0 entity_id: buster id or ghost id
pos=(1,2)# 1,2 x,y: position of this buster / ghost
e_type=3# 3 entity_type: the team id if it is a buster, -1 if it is a ghost.
state=4# 4 state: For busters: 0=idle, 1=carrying a ghost.
value=5# 5 value: For busters: Ghost id being carried. For ghosts:  of busters attempting to trap

# game loop
while True:
    entities_n = int(input())  # the number of busters and ghosts visible to you
    entity=[]
    for i in range(entities_n):
        element=[int(j) for j in input().split()]
        if element[e_type]==my_team_id:
            my_buster.append(element)
        if element[e_type]==-1:
            ghost.append(element)
    #print("busters:",my_buster,file=sys.stderr)
    #print("ghosts:",ghost,file=sys.stderr)
    for buster in my_buster:
        if buster[state]==1:
            if dist(buster,[0,home_x,home_y])<1600:
                print("RELEASE")
            else:
                print("MOVE",home_x,home_y)
        elif len(ghost)!=0:
            for coso in ghost:
                if dist(buster,coso)<1760 and coso[value]==0:
                    print("BUST",coso[entity_id])
                else:
                    print("MOVE",coso[1],coso[2])
                    print(coso[0],file=sys.stderr)
        else:#elif buster[state]==0:
            #angle=90/len(my_buster)
            print("MOVE",p_x,p_y)
        #print(,file=sys.stderr)

        # MOVE x y | BUST id | RELEASE

File: CodeBusters.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

//id,x,y,type,state,value
class Entity {
public:
    static int teamId;
    int id;
	int x;
	int y;
	int type;
	int state;
	int value;
	Entity (int a, int b):x(a),y(b){}
	Entity () {}
    Entity (Entity * copied) {
        this->id = copied->id;
		this->x = copied->x;
		this->y = copied->y;
		this->type = copied->type;
		this->state = copied->state;
		this->value = copied->value;
    }
    int entDist(Entity b){
		return sqrt((this->x - b.x)*(this->x-b.x)+(this->y-b.y)*(this->y-b.y));
	}
	void print(){
        if (this->type==this->teamId){
			cerr<<"Buster #"<<this->id<<" in "<<this->x<<", "<<this->y<<";";
			if (this->state==1){
				cerr<<" carrying ghost #"<<this->value<<endl;
			}else{
				cerr<<" not carrying ghosts"<<endl;
			}
		}else{
			cerr<<"Ghost #"<<this->id<<" in "<<this->x<<", "<<this->y<<";";
			if (this->value!=0){
				cerr<<" currently targeted by "<<this->value<<" busters"<<endl;
			}
		}
	}
	void copy (Entity copied) {
		this->id=copied.id;
		this->x=copied.x;
		this->y=copied.y;
		this->type=copied.type;
		this->state=copied.state;
		this->value=copied.value;
        }
};
int Entity::teamId;
int main(){
    int bustersPP,ghostCount,teamId; // # of busters * player ; # of ghosts in map; player 0 or 1;
    cin >> bustersPP >> ghostCount >> teamId; cin.ignore();
    cerr<<bustersPP<<" "<<ghostCount<<" "<<teamId<<endl;
	int baseX,baseY,destX,destY;//assign base and (first try) direction
	Entity base,dest;
	Entity::teamId=teamId;
	if (teamId==0){
		base = new Entity (0,0);
		dest = new Entity (16001,9001);
	}else{
		base = new Entity (16001,9001);
		dest = new Entity (0,0);
	}
	
    // game loop
    while (1) {
		Entity buster[bustersPP];
		Entity ghost[ghostCount];
        Entity temp = new Entity ();
        int busters=0,ghosts=0,entN;
        cin >> entN; cin.ignore();
        //input
        for (int i = 0; i < entN; i++){
            cin>>temp.id>>temp.x>>temp.y>>temp.type>>temp.state>>temp.value;cin.ignore();
			if (temp.type==teamId) buster[busters]=new Entity(temp);
            else if(temp.type==-1) ghost[ghosts]=new Entity(temp);
            
        }
        /*debug*/
		cerr<<"# busters:"<<busters<<endl;
		for (int i = 0;i < busters;i++){
			buster[i].print();
		}
		cerr<<"# ghosts:"<<ghosts<<endl;
		for (int i = 0;i < ghosts;i++){
			ghost[i].print();
		}
		/*debug*/
        //operations
        // MOVE x y | BUST id | RELEASE
 		for (int i = 0;i < busters;i++){
			if (buster[i].state==1){
				if (buster[i].entDist(base)<1600){
					cout<<"RELEASE"<<endl;
				}else{
					cout<<"MOVE "<<base.x<<" "<<base.y<<" home " << buster[i].value<<endl;
				}
			}
			else{
				bool found=false;
				for (int j = 0;j < ghosts;j++){
					if (ghost[j].value!=0){continue;}
					int dist = buster[i].entDist(ghost[j]);
					if (dist<1760){
						cout<<"BUST "<<ghost[j].id<<endl;
				    	found=true;
				    }else if (dist<2200){
				    	cout<<"MOVE "<<ghost[j].x<<" "<<ghost[j].y<<", "<<ghost[j].value<<" found"<<endl;
				    	found=true;
					}
				}
				if (!found){
					cout<<"MOVE "<<dest.x<<" "<<dest.y<<" nobody"<<endl;
				}
			}
		}
    }
}

